app/code/core/Mage/Eav/Model/Config.php

```diff
@@ -571,7 +571,9 @@ public function getEntityAttributeCodes($entityType, $object = null)
             }
             return $attributeCodes;
         } else {
-            return array_keys($this->_entityTypeAttributeIdByCode[$storeId][$entityType->getId()]);
+            return isset($this->_entityTypeAttributeIdByCode[$storeId][$entityType->getId()])
```

Comment:
Wouldnt `return array_keys($this->_entityTypeAttributeIdByCode[$storeId][$entityType->getId()] ?? [])` do the same?

Reply:
Changing array_keys to isset changes the return of the function  which is supposed to return attribute codes, not a boolean value.

Reply:
@davidhiendl It doesnt return boolean. I think you read that wrong. :)

Reply:
Yeah that code comment snippet function did cut of mid-change... fabulous.

Reply:
@pquerner I think `array_keys($this->_entityTypeAttributeIdByCode[$storeId][$entityType->getId()] ?? [])` would still trigged the undefined array index warning

Reply:
@fballiano https://onlinephp.io/c/eebcc nope, unless I did something wrong.

Reply:
`array_keys($this->_entityTypeAttributeIdByCode[$storeId][$entityType->getId()] ?? [])` works but not necessary better.

My break down:
1. The null coalescing operator (??) doc is [here](https://www.php.net/manual/en/migration70.new-features.php#migration70.new-features.null-coalesce-op). `$result = $array['a'] ?? []` is the same as `$result = isset($array['a']) ? $array['a'] : []`
2. after the above, we have `array_kyes($result)`, an additional step if key 'a' is not set.

This `isset($array['a']) ? array_keys($array['a']) : []` works slightly better if key 'a' is not set. However, this statement `array_keys($array['a'] ?? [])` is shorter.

I vote for the lengthy one: it has clearer intent and works slightly better if the key is not set. Having said that, I am fine  either one.